tree/tree/inc/ROOT/RFriendInfo.hxx

```diff
@@ -58,13 +63,30 @@ struct RFriendInfo {
     */
    std::vector<std::vector<std::int64_t>> fNEntriesPerTreePerFriend;
 
-   void AddFriend(const std::string &treeName, const std::string &fileNameGlob, const std::string &alias = "");
+   /**
+      Information on the friend's TTreeIndexes.
+   */
+   std::vector<std::unique_ptr<TVirtualIndex>> fTreeIndexInfos;
+
+   RFriendInfo() = default;
+   RFriendInfo(const RFriendInfo &);
+   RFriendInfo &operator=(const RFriendInfo &);
```

Comment:
This has become a heavy struct, I wonder if we can avoid enabling copy semantics. In what case do we need to copy the friend info around once created?

Reply:
Users of the struct should take care of moving instead of copying when possible (as it happens commonly with user-defined types), but I don't think it's the responsibility of the struct to completely disallow copies. Also it would be a backward-incompatible change as RFriendInfo is in the public namespace -- I wouldn't mix it with this PR which is only meant to fix #12260 .

Reply:
Agreed!